python-sdk/tests/files/locations/test_sftp.py

```diff
@@ -0,0 +1,80 @@
+import json
+from unittest.mock import patch
+
+import pytest
+from airflow.models.connection import Connection
+from airflow.providers.sftp.hooks.sftp import SFTPHook
+
+from astro.exceptions import PermissionNotSetError
+from astro.files.locations import create_file_location
+
+
+@patch("airflow.providers.sftp.hooks.sftp.SFTPHook.get_connection")
+def test_get_transport_params_for_sftp(mock_sftp_hook):
+    """test get_transport_params() method which should return connect_kwargs when keyfile is passed"""
+    mock_sftp_hook.return_value = Connection(
+        conn_id="sftp_default",
+        conn_type="test",
+        login=1234,
+        host="localhost",
+        extra=json.dumps({"key_file": "/some/local/path/rsa.pem"}),
+    )
+    path = "sftp://bucket/some-file"
+    location = create_file_location(path)
+    credentials = location.transport_params
+    assert credentials == {"connect_kwargs": {"key_filename": "/some/local/path/rsa.pem"}}
+
+
+@patch("airflow.providers.sftp.hooks.sftp.SFTPHook.get_connection")
+def test_get_transport_params_for_sftp_no_value(mock_sftp_hook):
+    """test get_transport_params() method when no keyfile is passed"""
```

Comment:
I think doc-string doesn't reflect the test.

Reply:
changed to test get_transport_params() method raises exception when no credentials is passed